NIFI-5055 added ability to unpenalize MockFlowFile directly or from M…

markobean opened the pull request:
…ockProcessSession

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

ottobackwards commented:
Looks good to me

MikeThomsen commented on nifi-mock/src/test/java/org/apache/nifi/util/TestMockProcessSession.java:
```diff
@@ -101,6 +101,17 @@ public void testKeepPenalizedStatusAfterPuttingAttribute(){
+        final MockProcessSession session = new MockProcessSession(new SharedSessionState(processor, new AtomicLong(0L)), processor);
+        FlowFile ff1 = session.createFlowFile("hello, world".getBytes());
+        ff1 = session.penalize(ff1);
+        assertEquals(true, ff1.isPenalized());
```
A little better if it's `assertTrue`

MikeThomsen commented on nifi-mock/src/test/java/org/apache/nifi/util/TestMockProcessSession.java:
```diff
@@ -101,6 +101,17 @@ public void testKeepPenalizedStatusAfterPuttingAttribute(){
+        ff1 = session.penalize(ff1);
+        assertEquals(true, ff1.isPenalized());
+        ff1 = session.unpenalize(ff1);
+        assertEquals(false, ff1.isPenalized());
```
Please change to `assertFalse`

markobean commented:
Updated to use assertTrue and assertFalse. I had simply borrowed assertEquals from the previous unit test. So, while updating, I updated that test to use assertTrue as well. Commit, squashed and pushed.

MikeThomsen commented:
@markobean You close the PR now. I merged the change.

pvillard31 commented:
@MikeThomsen - just fyi, while merging a PR it's highly recommended to amend the commit message to include the magic words that will automatically close the github PR. The commands I usually execute when merging a PR are:

````
git checkout master
git pull origin master
git fetch github
git checkout github/pr/<PR ID>
git checkout -b pr<PR ID>
--- here you can do additional reviewing, maven builds, test with a running instance, etc ---
git commit --amend -s
--- you update the commit message to include "This closes #<PR ID>."
git log
--- the above command is to retrieve the commit ID ---
git checkout master
git cherry-pick <commit ID>
git push origin master
````

This is an example - there are probably other ways to do it.

MikeThomsen commented:
@pvillard31 Thanks. I finally figured it out after doing a second one today.

markobean commented:
So, does this PR need to be manually closed with the "Close and comment" button?

pvillard31 commented:
That's correct @markobean - it has been merged into master with 274ed46d923e7d080b57ed121596b4ecfa997a2c.

markobean commented:
Changes committed to master. Closing PR.
